engine/consensus/ingestion/engine.go

```diff
@@ -197,6 +197,15 @@ func (e *Engine) onGuarantee(originID flow.Identifier, guarantee *flow.Collectio
 		return nil
 	}
 
+	// don't propagate collection guarantees if we are not currently staked
+	staked, err := protocol.IsNodeStakedAt(e.state.Final(), e.me.NodeID())
```

Comment:
We called `e.state.Final()` a few times within this function, should we lock the snapshot of the finalized block, in case it shifts, which I'm afraid is hard to debug.